src/app/ui-components/selectFeeRate/feeSelectPopup.tsx

```diff
@@ -92,7 +91,11 @@ function FeeSelectPopup({
   baseToFiat,
   setFeeRate,
   getFeeForFeeRate,
+  absoluteBalance,
 }: Props) {
+  const stxInputExtractor = /[0-9]+[.]?[0-9]{0,6}/;
+  const inputValidator = absoluteBalance ? stxInputExtractor : /^[0-9]*$/;
```

Comment:
sx specific code shouldn't live in this generic component. It also shouldn't rely on a non-stx related variable (absoluteBalance) to enable stx functionality. This should live in the outer, stx wrapper for this component.

The absoluteBalance arg also seems to be added in a non-generic way and the implementation seems to be specific to stx. I'm pretty sure this can also be removed in favor of the getFeeForFeeRate function returning undefined when there is not enough balance, as is done with the btc implementation.

Reply:
I think I did it this way because the BTC implementation could calculate the fees outside of the component with a transaction summary, do we have a similar analog for STX?